node/src/filecheckpointer.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import fs from 'fs';
+import { Checkpointer } from './checkpointer';
+
+export interface CheckpointerState {
+  blockNumber?: string;
+  transactionIDs: string[];
+}
+
+export class FileCheckPointer implements Checkpointer {
+    #path: string;
+    #blockNumber?: bigint;
+    #transactionIDs: Set<string> = new Set();
+
+    constructor(path: string) {
+        this.#path = path;
+    }
+
+    async init(): Promise<void> {
+        await this.#getStateFromFile();
+        await this.#saveDate();
+    }
+
+    async checkpoint(blockNumber: bigint, transactionId?: string): Promise<void> {
+        if (blockNumber !== this.#blockNumber) {
+            this.#blockNumber = blockNumber;
+            this.#transactionIDs.clear();
+        }
+        if (transactionId) {
+            this.#transactionIDs.add(transactionId);
+        }
+        await this.#saveDate();
+    }
+
+    getBlockNumber(): bigint | undefined {
+        return this.#blockNumber;
+    }
+
+    getTransactionIds(): Set<string> {
+        return this.#transactionIDs;
+    }
+
+    async #getStateFromFile(): Promise<void> {
```

Comment:
getters usually return some value. Perhaps a name like `load()` (and changing `saveData()` to be just `save()` to mirror it) would be clearer? Or if you want to be more verbose maybe `loadFromFile()` and `saveToFile()`?